### ISSUE TITLE
Perplexity plaeau when training embeddings

### ISSUE THREAD

**Issue by eh-93** [flairNLP/flair#2190]
Hello

I'm training a forward model on a fairly sizeable dataset. For the past few days the perplexity seems to have stalled at ~5.
Just wondering if anyone has experienced this and if there's anything I can do about it or is it just a matter of training on more epochs? 

My corpus is around 100GB of short sentences, training parameters:

```
#Params
is_forward_lm = True
hidden_size = 2048
nlayers = 1
sequence_length = 250
mini_batch_size = 216
max_epochs = 1

# Inputs
corpus_path = "ml/datasets/corpus"
char_file = f"{corpus_path}/char_mappings"
model_name = "forward_model" if is_forward_lm else "backward_model"
output_path = f"ml/models/{model_name}"
new_model = False if os.path.exists(f"{output_path}/checkpoint.pt") else True

# Character embeddings
dictionary = Dictionary.load_from_file(char_file)
corpus = TextCorpus(
    Path(corpus_path), 
    dictionary, 
    is_forward_lm, 
    character_level=True
)

language_model = LanguageModel(
    dictionary, 
    is_forward_lm, 
    hidden_size=hidden_size, 
    nlayers=nlayers
)

trainer = LanguageModelTrainer(language_model, corpus)

if new_model:
    print("Training new model")
    trainer.train(
        output_path,
        sequence_length=sequence_length,
        mini_batch_size=mini_batch_size,
        max_epochs=max_epochs,
        checkpoint=True,
        use_amp=True
    )
else:
    print("Resuming Training")
    checkpoint_path = Path(f"{output_path}/checkpoint.pt")
    trainer = LanguageModelTrainer.load_from_checkpoint(checkpoint_path, corpus)
    trainer.train(
        output_path,
        sequence_length=sequence_length,
        mini_batch_size=mini_batch_size,
        max_epochs=max_epochs,
        checkpoint=True,
        use_amp=True
    )

```

Last split:
2021-03-28 09:34:57,648 | end of split 7073 /8557 | epoch   1 | time: 42.82s | valid loss  1.62 | valid ppl     5.03 | learning rate 0.0000
